### DISCUSSION TITLE
How to distinguish provider stations

### DISCUSSION THREAD

**Discussion by mmuffins** [Opinionated-Blueprints/10-Books-Full-of-Rails#22] [Q&A] [answered]
Hi,
I'm implementing the train system in my game, and set up a small test area using a mix of blueprint books v2.7.0 and v2.6.0 (because 2.7.0 removed trains and rails). Everything works perfectly, the only thing I can't quite figure out is how to tell provider stations apart.

In the screenshots below I created a basic infrastructure with rails, depots, and fuel supply, and then added three provider - requester station pairs for iron, copper, and uranium respectively. All stations and trains were set to use group 0.

![train_overview](https://github.com/user-attachments/assets/67ab6de3-1efe-401d-b00e-6176b7370b48)
![overview](https://github.com/user-attachments/assets/4e077bd4-3716-4b98-803b-6d085764677a)

The trains choose random stations to pick up and deliver resources, and everything gets delivered, so I assume I configured it correctly. But from the map I cannot tell the provider stations apart since they all use the item parameter icon in their station name. 

Is there a trick to do this?

**Comment by rpdelaney** [comment]
> The trains choose random stations to pick up and deliver resources, and everything gets delivered, so I assume I configured it correctly.

So... what's the problem?

**Reply by mmuffins** [reply]
> So... what's the problem?

I'm wondering if I'm doing something wrong because all the provider stations have the same name. It's not a problem for my small test area in the screenshot, but with bigger factories it might get confusing if all provider stations have the same name.

**Comment by Bocian-1** [comment]
With this system, you can't. It relies on providers having identical names to work. In that sense, it's just a more generalised version of the system we've had in 1.1, but instead of having identical providers per-resource, we have all providers being identical and it still works. (I may be able to make it viable to make a 1.1-like system with trains per resource instead of completely generic sometime in the future)

**Comment by Bocian-1** [comment, marked as answer]
Added display panels to provider stations in v2.12.0. They are now displaying the provided resource in the map view.

**Reply by mmuffins** [reply]
Amazing, thanks a lot!
